src/gui/MainWindow.cc

```diff
@@ -2257,7 +2257,8 @@ void MainWindow::sendToLocalSlicer()
     exportFileFormat = FileFormat::STL;
   }
 
-  const auto tmpPath = QDir::temp().filePath("OpenSCAD.XXXXXX."+fileFormat.toLower());
+  QString basename = QString(std::filesystem::path(activeEditor->filePath.toStdString()).stem().c_str());
```

Comment:
The `filePath` is empty if the file is not saved yet. Currently this is handled in various places, so we either need to do that here too, or have some dedicated method getting "name or *Untitled*" similar to "exportPath()".

Reply:
I'm doing some related refactoring in #5357 - will fold this feature into that PR